### PR TITLE
Add hook for creating lobby before participants join

### PR DESCRIPTION
**Pull request by abora8x8**

### CODE REVIEW

**Comment by saghul** [comment]
Please be more descriptive in both your commit description and extended message. As it stands I have no idea what this does.

You should write so that even a person who is not deep into that part of the codebase can get a glimpse of what you are trying to do.

**Comment by abora8x8** [comment]
In muc_password_present there is an external call to get the password, moderator id and the fact that a room has lobby enabled before the meeting start. If that is the case the "create-lobby-room" event will be fired.

File: resources/prosody-plugins/mod_muc_lobby_rooms.lua [hunk 1]
```diff
@@ -373,6 +377,23 @@ function update_session(event)
     end
 end
 
+function handle_create_lobby(event)
```
Comment (damencho) [inline resources/prosody-plugins/mod_muc_lobby_rooms.lua:380]: Can you change the 'muc-config-submitted' hook to use this method, so we avoid code duplication?
Maybe make the function return true/false so you can act whether lobby room was created or not (the notification and the `event.status_codes['104'] = true;` code)

Comment (damencho) [reply resources/prosody-plugins/mod_muc_lobby_rooms.lua:380]: Also do the save room only if room was created.

**Comment by saghul** [comment]
> If you want to enable lobby before the meeting starts you can do so by firing the "create-lobby-room" event.

Please put that in the commit message description.

File: resources/prosody-plugins/mod_muc_lobby_rooms.lua (hunk 1 above)
Comment (abora8x8) [reply resources/prosody-plugins/mod_muc_lobby_rooms.lua:389]: Done.

**Pull request merged**